packages/integration_test/ios/Classes/IntegrationTestPlugin.m

```diff
@@ -45,19 +45,9 @@ - (instancetype)init {
 }
 
 + (void)registerWithRegistrar:(NSObject<FlutterPluginRegistrar> *)registrar {
-  // No initialization happens here because of the way XCTest loads the testing
-  // bundles.  Setup on static variables can be disregarded when a new static
-  // instance of IntegrationTestPlugin is allocated when the bundle is reloaded.
-  // See also: https://github.com/flutter/plugins/pull/2465
-}
-
-- (void)setupChannels:(id<FlutterBinaryMessenger>)binaryMessenger {
-  FlutterMethodChannel *channel =
-  [FlutterMethodChannel methodChannelWithName:kIntegrationTestPluginChannel
-                              binaryMessenger:binaryMessenger];
-  [channel setMethodCallHandler:^(FlutterMethodCall *call, FlutterResult result) {
-    [self handleMethodCall:call result:result];
-  }];
+  FlutterMethodChannel *channel = [FlutterMethodChannel methodChannelWithName:kIntegrationTestPluginChannel
+                                                              binaryMessenger:registrar.messenger];
+  [registrar addMethodCallDelegate:[self instance] channel:channel];
```

Comment:
For future archeologists: The `addMethodCallDelegate` does the `setMethodCallHandler` logic that was originally used, so this is semantically the same as the previous `setMethodCallHandler` call.